gax-java/gax/src/main/java/com/google/api/gax/rpc/ClientContext.java

```diff
@@ -178,6 +178,7 @@ public static ClientContext create(StubSettings settings) throws IOException {
 
     String settingsGdchApiAudience = settings.getGdchApiAudience();
     Credentials credentials = settings.getCredentialsProvider().getCredentials();
+    String credentialType = credentials.getCredentialType();
```

Comment:
What are the possible credential types? cc: @ldetmer as it may affect the API key project.

Reply:
current metric design includes 4 types: u - UserCredentials, sa - ServiceAccountCredentials, mds - ComputeEngineCredentials, imp - ImpersonatedCredentials